iothub/device/src/Pipeline/RetryDelegatingHandler.cs

```diff
@@ -10,26 +10,37 @@
 
 namespace Microsoft.Azure.Devices.Client.Transport
 {
+    internal enum ClientTransportStatus
+    {
+        Closed = 0,
+        Open = 1,
+    }
+
     internal sealed class RetryDelegatingHandler : DefaultDelegatingHandler
     {
         // RetryCount is used for testing purpose and is equal to MaxValue in prod.
         private const uint RetryMaxCount = uint.MaxValue;
 
+        private readonly SemaphoreSlim _clientOpenSemaphore = new(1, 1);
+        private readonly SemaphoreSlim _cloudToDeviceMessageSubscriptionSemaphore = new(1, 1);
+        private readonly SemaphoreSlim _directMethodSubscriptionSemaphore = new(1, 1);
+        private readonly SemaphoreSlim _twinEventsSubscriptionSemaphore = new(1, 1);
+
         private readonly RetryHandler _internalRetryHandler;
         private IIotHubClientRetryPolicy _retryPolicy;
 
-        private bool _isOpen;
-        private SemaphoreSlim _handlerSemaphore = new(1, 1);
         private bool _methodsEnabled;
         private bool _twinEnabled;
         private bool _deviceReceiveMessageEnabled;
+        private bool _isDisposing;
+        private long _clientTransportState; // references the current client transport status as the int value of ClientTransportStatus
```

Comment:
rename to align with other places?

```suggestion
        private long _clientTransportStatus; // references the current client transport status as the int value of ClientTransportStatus
```